include/tvm/runtime/packed_func.h

```diff
@@ -994,6 +1008,22 @@ inline TVMType String2TVMType(std::string s) {
     t.bits = 1;
     t.lanes = 1;
     return t;
+  } else if (s.substr(0, 6) == "custom") {
+    // TODO(gus): too much hardcoding here.
+    scan = s.c_str() + 6;
+    if (*scan != '[')
+      LOG(FATAL) << "expected opening brace after 'custom' type in" << s;
+    ++scan;
+    size_t custom_name_len = 0;
+    while (scan + custom_name_len <= s.c_str() + s.length() &&
+           *(scan + custom_name_len) != ']')
+      ++custom_name_len;
+    if (*(scan + custom_name_len) != ']')
+      LOG(FATAL) << "expected closing brace after 'custom' type in" << s;
+    scan += custom_name_len + 1;
+
+    auto type_name = s.substr(7, custom_name_len);
+    t.code = GetCustomTypeCode(type_name);
```

Comment:
maybe using a regex here would be more readable?

Reply:
this code segment is difficult to decipher, agreed with @weberlo. Or perhaps add a comment on what's going on.

Reply:
We should at least separate into a parsing function, and replace now or later.

Reply:
Went with Jared's approach. Didn't want to bring in some regex dependency, though regex would surely be clearer. Moved to a TVM_DLL function hidden in the implementation.

Reply:
My implementation caused a bug so I reverted it -- will worry about it later.

Reply:
Alright, fixed now.